lib/src/validator/sdk_constraint.dart

```diff
@@ -20,15 +20,25 @@ class SdkConstraintValidator extends Validator {
 
   Future validate() async {
     var dartConstraint = entrypoint.root.pubspec.dartSdkConstraint;
-    if (dartConstraint is VersionRange &&
-        dartConstraint.toString().startsWith("^")) {
-      errors
-          .add("^ version constraints aren't allowed for SDK constraints since "
-              "older versions of pub don't support them.\n"
-              "Expand it manually instead:\n"
-              "\n"
-              "environment:\n"
-              "  sdk: \">=${dartConstraint.min} <${dartConstraint.max}\"");
+    if (dartConstraint is VersionRange) {
+      if (dartConstraint.toString().startsWith("^")) {
+        errors.add(
+            "^ version constraints aren't allowed for SDK constraints since "
+            "older versions of pub don't support them.\n"
+            "Expand it manually instead:\n"
+            "\n"
+            "environment:\n"
+            "  sdk: \">=${dartConstraint.min} <${dartConstraint.max}\"");
+      }
+
+      if (dartConstraint.max == null) {
```

Comment:
So we require an upper bound if they have an SDK constraint, but we still allow completely omitting the SDK constraint. Should we require having one?

Reply:
We shouldn't allow a completely omitted one either, I will add an additional test to verify that behavior.

Reply:
test added